public/app/plugins/datasource/cloud-monitoring/components/Metrics.tsx

```diff
@@ -109,6 +110,8 @@ export function Metrics(props: Props) {
         label: m.displayName,
         description: m.description,
       }));
+    // On service change reset all query values except the project name
+    Object.assign(query, { ...defaultTimeSeriesList(datasource), projectName: query.projectName });
```

Comment:
Updating the service value triggers `onChange` as it causes the `onMetricTypeChange` function to be called 😊

Reply:
mmh, that would be relying on an indirect reference. `onMetricTypeChange` is getting the `query` object from `VisualMetricQueryEditor`, not from this component so it's a miracle if that works :sweat_smile:

I would move the definition of `onMetricTypeChange` to this component (since it's only used here) and explicitly build a full `query` object here that is used for the main `onChange` so it's explicit.